sdk/src/cameras/itof-camera/camera_itof.cpp

```diff
@@ -1190,27 +1193,31 @@ void CameraItof::freeConfigData(void) {
 
 aditof::Status CameraItof::readSerialNumber(std::string &serialNumber,
                                             bool useCacheValue) {
-    aditof::Status stat;
-    if (useCacheValue == false) {
-        stat = readAdsd3500CCB();
-        serialNumber = m_details.serialNumber;
-    } else {
-        serialNumber = m_details.serialNumber;
-        if (serialNumber == "")
-            stat = aditof::Status::UNAVAILABLE;
-        else
-            stat = aditof::Status::OK;
+    using namespace aditof;
+    Status status = Status::OK;
+
+    if (useCacheValue) {
+        if (m_details.serialNumber != "") {
```

Comment:
you can check is a string is empty or not. It think the method is called Emtpy()